### PR TITLE
[CDAP-7554] Add HTTP endpoint to list program runs for an app version

### PR DESCRIPTION
**Pull request by ecfm**
https://issues.cask.co/browse/CDAP-7554
http://builds.cask.co/browse/CDAP-DUT5014-5

### CODE REVIEW

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java [hunk 1]
```diff
@@ -687,8 +687,10 @@ protected void stopProgram(ProgramId program, String runId, int expectedStatusCo
       path = String.format("apps/%s/versions/%s/%s/%s/stop", program.getApplication(), program.getVersion(),
                            program.getType().getCategoryName(), program.getProgram());
     } else {
-      path = String.format("apps/%s/%s/%s/runs/%s/stop", program.getApplication(),
-                           program.getType().getCategoryName(), program.getProgram(), runId);
+      // TODO: HTTP endpoint for stopping a program run of an app version not implemented
```
Comment (gokulavasan) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java:690]: Hmm, do we need to implement it? Don't we have that for services?

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java [hunk 2]
```diff
@@ -687,8 +687,10 @@ protected void stopProgram(ProgramId program, String runId, int expectedStatusCo
...
+      path = null;
+//      path = String.format("apps/%s/%s/%s/runs/%s/stop", program.getApplication(),
```
Comment (gokulavasan) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java]: Remove commented code.

**Comment by bdmogal** [comment]
what's the use-case for adding a new REST API to 3.6? Ideally, since 3.6 has been released, only bugfix/improvements should go in there. Has there been a specific request for this API in 3.6?

**Comment by sreevatsanraman** [comment]
Agree to @bdmogal. We should open this against develop.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java (hunk 1 above)
Comment (ecfm) [reply cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java:690]: We don't have method to start/stop a program given specific runId.

Comment (gokulavasan) [reply cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java:690]: Ah I see ok. Can you modify the comment to specify that it is for a specific runId. Also, open a JIRA for the same.

**Comment by gokulavasan** [comment]
LGTM 👍

**Pull request merged**